### ISSUE TITLE
Infra: Integrate Ruff

### ISSUE THREAD

**Issue by Goldziher** [starlite-api/starlite#858]
We want to integrate `ruff` into our pre-commit config, and in general improve our `pre-commit` setup.

**Comment by cofin** [comment]
I left this note on the actual PR, but it probably should have been added here:

So we do have a few plugins that haven't been incorporated into ruff yet. Here is the summary of my testing:

What can be replaced with ruff:

pycodestyle
pyupgrade
flake8
flake8-bugbear
flake8-comprehensions
flake8-mutable
flake8-print
isort *I don't see a way to support multi_line_output = 3, but we may not need it. I will open an issue with the ruff project to see if there is a way to enable or if it's on the roadmap.
The following specific checks aren't currently supported.

SCS108 from 'flake8-secure-coding-standard'
E225, W503, E203 from pycodestyle
The following utilities are not handled by ruff:

flake8-noqa
flake8-return
flake8-simplify
flake8-type-checking
flake8-implicit-str-concat
flake8-secure-coding-standard
flake8-encodings
flake8-use-fstring
flake8-use-pathlib
flake8-mutable
I've currently left all of the steps in the pre-commit, and I've also added the ruff check.

What do you guys think the best way to deal with this is?

**Comment by Goldziher** [comment]
The only two flake8 plugins we really have to have are these: 

flake8-simplify
flake8-type-checking

We can drop the others. If ruff supports the functionality of these two, we can simply switch to flake8. Until then, i don't really see the advantage of using ruff instead of flake8 - if we anyhow have to run it. Additionally we are running flake8 on docstrings. Does ruff give us this functionality?

**Comment by cofin** [comment]
No, it doesn't have the flake8 docstring support yet, to my knowledge.  

I have to agree, swapping to ruff with it's current functionality doesn't make a lot of sense.  I think we should continue to monitor the new features and revisit this when it covers a more exhaustive list of modules we use.

**Comment by Goldziher** [comment]
Aight, lets close this for now.
